src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -131,6 +132,39 @@ public void assignTaskToPerson(StudentId studentId, Task task) {
         }
     }
 
+    /**
+     * Assigns {@code task} to a person taking a module with {@code moduleId}.
+     *
+     * @param moduleCode the module code of the module of which all students are to be assigned a task.
+     * @param task the task to be assigned.
+     */
+
+    public void assignTaskToAllInModule(ModuleCode moduleCode, Task task) {
+        requireAllNonNull(moduleCode, task);
+        boolean anyPersonFound = false;
+        int totalPersonWithNoDuplicateTask = 0;
+        int totalPersonTakingThisModule = 0;
+
+        for (Person currPerson: internalList) {
+            if (currPerson.getModuleCode().equals(moduleCode)) {
+                anyPersonFound = true;
+                totalPersonTakingThisModule++;
+                if (!currPerson.isTaskAlreadyPresent(task)) {
+                    Person updatedPerson = currPerson.getCopy();
+                    updatedPerson.addTask(task);
+                    setPerson(currPerson, updatedPerson);
+                    totalPersonWithNoDuplicateTask++;
+                }
+            }
+        }
+        if (totalPersonTakingThisModule != totalPersonWithNoDuplicateTask) {
+            throw new DuplicateTaskException();
+        }
```

Comment:
For this part, the DuplicateTaskException is thrown if, for example, two people are taking CS2100, one person has a task "Homework1" while the other does not, and you attempt to do `assign m/CS2100 tn/Homework1`. The task is still successfully assigned, but the GUI will show "This task is already assigned to a specified person." Would it be better if you handled this case by writing how many people were assigned the task out of all who are taking the module? Probably many ways to handle this tbh.

Reply:
Alright, I made another exception to catch in the try block in AssignCommand, but it will return a CommandResult instead of throwing an exception